packages/cli/src/PublicApi/v1/handlers/executions/executions.handler.ts

```diff
@@ -37,6 +38,11 @@ export = {
 				executionId: execution.id,
 			});
 
+			Container.get(ConcurrencyControlService).remove({
```

Comment:
I have just realized there's a hole here, we're not checking whether the execution is still running. We simply delete it from DB regardless.

If it's still running, this is going to cause some uncaught errors down the line, where the execution would finish and try to update the db but fail. Maybe outside of the scope of this PR.

Also when removing it from the concurrency control, we gotta check this first. If it's queued, we should dequeue first, so it won't start, and then hard delete from the DB if we successfully removed it from the queue right? I think that if it's running, we might simply return a 400 in this endpoint saying the execution is in progress. Perhaps create a separate endpoint to stop executions, idk.

Reply:
>If it's queued, we should dequeue first, so it won't start, and then hard delete from the DB if we successfully removed it from the queue right?

I agree about the order (dequeue then delete) but I don't think we should make the deletion conditional on dequeuing success, else we would be disallowing deletion of completed executions. Changed the order and made dequeueing conditional on execution status. Please let me know that you think.

Reply:
> else we would be disallowing deletion of completed executions

So if the execution is not found in the queue we should proceed to regular deletion, we should only go this path if an execution risks being in the queue or currently executing, that's what I meant.

Reply:
As discussed during our sync, this should cover it:

```ts
if (execution.status === 'running') {
	return res.status(400).json({
		message: 'Cannot delete a running execution',
	});
}

if (execution.status === 'new') {
	Container.get(ConcurrencyControlService).remove({
		executionId: execution.id,
		mode: execution.mode,
	});
}

await Container.get(ExecutionRepository).hardDelete({
	workflowId: execution.workflowId,
	executionId: execution.id,
});
```